[Leaderboards] Landing Page Views

jgb4 opened the pull request:
Wrote view, URL, and template for the leaderbaords landing page following [this](https://docs.google.com/document/d/1gf-WN8yMf89U6clCSZm9htkQrlEzh_kxItrSXfIQit8/edit?tab=t.0) style. Landing page currently does not dsiplay anything other than text for logged out users. Also does not currently support switching between settings (default view metric and anonymity), although the template leaves space for these dropdowns. To properly implement settings, we may need to add them as attributes to the User or UserProfile model (cc User team)


I did not mean to commit changes to the knowledge_base file. Will have to exclude that from PR somehow (cc Chis)

Commit:
Wrote view, URL, and template for the leaderbaords landing page. Currently does not dsiplay anything other than text for logged out users. Also does not currently support switching between settings (default view metric and anonymity), although the template leaves space for these dropdowns. To properly implement settings, we may need to add them as attributes to the User or UserProfile model (cc User team)

Commit:
[Leaderboards] Make display name click into user profile

karunkundrum committed:
fixing extra file

karunkundrum commented:
I just deleted the extra file you added in the commit above to expedite this.

karunkundrum reviewed (approved):
This looks good!

jgb4 committed:
Merge branch 'dev' into jgb4-leaderboards-landing-page-pr3

jgb4 committed:
Resolve merge conflicts

jgb4 committed:
Update leaderboard.html

karunkundrum committed:
fix linter

karunkundrum commented:
I noticed that the linter wasn't passing so I just fixed it. It looks ready to be merged now!

jgb4 committed:
Merge branch 'jgb4-leaderboards-landing-page-pr3' into jgb4-leaderboards-user-profile-click

jgb4 committed:
Merge branch 'dev' into jgb4-leaderboards-landing-page-pr3

jgb4 committed:
Merge pull request #1161 from uchicago-cs/jgb4-leaderboards-user-profile-click

[Leaderboards] Make display name click into user profile

jgb4 committed:
Merge branch 'dev' into jgb4-leaderboards-landing-page-pr3

cmarshall282 commented on src/chigame/leaderboards/views.py:
```diff
@@ -32,3 +33,49 @@ def leaderboard_view(request, game_id):
+    anonymity = None
+
+    if request.user.is_authenticated:
+        user_profile = get_object_or_404(UserProfile, user=request.user)
```
We probably don't want to throw a 404 if the user doesn't have a profile. Can we handle this more gracefully?

cmarshall282 commented on src/templates/leaderboards/landing_page.html:
```diff
@@ -0,0 +1,31 @@
+          <tr>
+            <td>{{ entry.leaderboard.game.name }}</td>
+            <td>{{ entry.rank }}</td>
+            <td>{{ entry.user.display_name }}</td>
```
display_name no longer exists in the user model. Please update this to name

cmarshall282 commented on src/chigame/leaderboards/views.py:
```diff
@@ -32,3 +33,49 @@ def leaderboard_view(request, game_id):
+
+        # user = User.objects.get(pk=request.user.pk)
+        # default_view_metric = user.default_metric
+        # anonymity = user_profile.anonymity_setting
```
This should use the leaderboard settings models (it should not be a user_profile setting)

cmarshall282 reviewed (changes requested):
There are a few things that we can handle better in this PR

jgb4 committed:
Merge branch 'dev' into jgb4-leaderboards-landing-page-pr3

Commit:
I addressed the comments and made several changes: 1. I changed the landing page to display the leaders for each game, instead of the user's position. This way, the landing page is the same for all users (including logged-out users). 2. I modified the fixture to include a region (Midwest) for the three users, and made minor changes to the leaderboard view so that it would correctly display the ranked players and their regions. 3. I linked the leaderboard landing page to the Chigame home page via the nav bar on the top of the screen, so all users can access the landing page. I also linked each leaderboard to its entry on the landing page so users can access all leaderboards. Related further PRs will also address: light CSS styling for landing page and leaderboards, and back buttons for the leaderboards and landing page.

jgb4 committed:
Merge branch 'dev' into jgb4-leaderboards-landing-page-pr3

jgb4 commented:
I addressed the comments and made several changes: 

1. I changed the landing page to display the leaders for each game, instead of the user's position. This way, the landing page is the same for all users (including logged-out users). 
2. I modified the fixture to include a region (Midwest) for the three users, and made minor changes to the leaderboard view so that it would correctly display the ranked players and their regions. 
3. I linked the leaderboard landing page to the Chigame home page via the nav bar on the top of the screen, so all users can access the landing page. I also linked each leaderboard to its entry on the landing page so users can access all leaderboards. 
Related further PRs will also address: light CSS styling for landing page and leaderboards, and back buttons for the leaderboards and landing page.


**To test, do the following:**
make sure database models are migrated.
run python3 manage.py loaddata test_leaderboard_fixture.json
run python3 manage.py runserver
go to http://127.0.0.1:8000/
click on the 'Leaderboards' tab at the top of the page:
<img width="1511" alt="Screenshot 2025-05-24 at 10 49 01 PM" src="https://github.com/user-attachments/assets/c89da024-4b18-482e-8059-2d80735b38a7" />

it should take you to the landing page:
<img width="433" alt="Screenshot 2025-05-24 at 10 46 10 PM" src="https://github.com/user-attachments/assets/67869984-a8a3-4cab-820e-18e5f31f6dec" />

click on any of the Game/Leader/Region entries, e.g. Chess:
<img width="389" alt="Screenshot 2025-05-24 at 10 46 42 PM" src="https://github.com/user-attachments/assets/b2a5315f-aa2d-4c8d-b084-344139848c2d" />

jgb4 committed:
Remove empty view

jgb4 committed:
Merge branch 'dev' into jgb4-leaderboards-landing-page-pr3

cmarshall282 reviewed (approved):
Looks great!

cmarshall282 committed:
Merge branch 'dev' into jgb4-leaderboards-landing-page-pr3
